src/launcher_gui.js

```diff
@@ -21,89 +26,123 @@ app.on('second-instance', () => {
 });
 
 app.prependListener('ready', () => {
-	const { wizard } = require('./launcher_wizard');
-
-	const display = electron.screen.getPrimaryDisplay();
-	const sWidth = display.workAreaSize.width;
-	const width = 800;
-	const height = process.platform === 'win32' ? 418 : 380 + 8;
-
-	let windowOpts = {
-		x: (sWidth - width) / 2,
-		// y: (sHeight - height) / 2,
-		y: 100,
-		width: width,
-		height: height,
-		show: false,
-		icon: `${__dirname}/renderer/spring-icon.png`,
-		webPreferences: {
-			nodeIntegration: true,
-			enableRemoteModule: true,
-		},
-	};
-	windowOpts.resizable = false;
-	Menu.setApplicationMenu(null);
-	mainWindow = new BrowserWindow(windowOpts);
-
-	mainWindow.loadFile(`${__dirname}/renderer/index.html`);
-	if (isDev) {
-		// mainWindow.webContents.openDevTools();
-	}
+	// TODO: is there a better way to get yargs' positional args ? Did not find
+	// it ...
+	if (argv._[0]) {
+		// If a replay file was passed as argument to spring-launcher, then we
+		// don't open the GUI, we just start the replay.
+
+		// TODO: when we support downloading assets (map/game/engine) we might
+		// need to show the GUI anyway, to provide feedback on the
+		// downloads/ask for user confirmation.
+		const { launcher } = require('./engine_launcher');
+
+		const replayPath = decodeURI(url.parse(argv._[0]).pathname);
+
+		const engineName = replayPath.split('_').pop().slice(0, -5).toLowerCase();
```

Comment:
is engine name really only contained in the replay file name? can't `sdfz-demo-parser` be used to parse the actual engine? (is it contained in .sdfz?)

Reply:
Maybe it can, but I think parsing the replay to get the engine name is quite wasteful.

Reply:
I can write a "better" way to extract it from the filename if you want - like, use a more robust regex and put it in a function ?

Reply:
In the end I just removed the fragile extension stripping, using the path module instead for that. IMO it's good enough, tell me what you think

Reply:
I am still leaning towards using sdfz-demo-parser after checking whether file exists.

- I think the "wastefulness" you are referring to (by which I think you mean the extra time it'd take to parse it) is negligible, as the file will have to be loaded from disk anyway, we just cache it a bit earlier.
- It's a less fragile approach. If a user or script renames a replay file it should still be loadable.
- It's a way to verify the correctness of the file before trying to feed it to Spring. Trying to load an invalid replay file might have Spring spit out some unintelligible error.
- We'll have to do it anyway if we want to support downloading missing resources, which is the next step I think.

Reply:
All good points indeed. The thing is, sdzf-parser on my machine can take from a few hundred msecs to many seconds (on the order of 10 seconds for really big replays) to parse a replay file - so a long time. The current UI (or lack of it) won't work, because the user will think it crashed/it is not working. I think there are some options in the replay parser to make it faster, but I'd have to see which ones are good for this use case.

 I would be for:

1. Trying to parse the engine name out of the file name, with some more validation (check that it conforms to a given format).
2. If it fails, then use the sdfz parser, and show the GUI while parsing, so that if/when it takes ~10secs the user has some feedback.

Since this is significantly more work than what I did, if it's OK with you I'd prefer to do this in an additional PR.

Reply:
Saw this and added an option to DemoParser to skip the packet parsing. I initially decided against it because it technically means some of the data in the info object can be incorrect or missing, such as player faction or start position, but if you're just trying to get info directly from the demo header, such as engine version, then it should be fine.

`const parser = new DemoParser({ skipPackets: true });` in `sdfz-demo-parser` version `4.5.0`